cli/tools/registry/pm/outdated.rs

```diff
@@ -0,0 +1,607 @@
+    }
+  }
+
+  Ok(())
+}
+
+fn choose_new_version_req(
+  dep: &Dep,
+  resolved: Option<&PackageNv>,
+  latest_versions: &PackageLatestVersion,
+  update_to_latest: bool,
+  filter_set: &filter::FilterSet,
+) -> Option<VersionReq> {
+  let explicit_version_req = filter_set
+    .matching_filter(dep.alias.as_deref().unwrap_or(&dep.req.name))
+    .version_spec()
+    .cloned();
+
+  if let Some(version_req) = explicit_version_req {
+    if let Some(resolved) = resolved {
+      // todo(nathanwhit): handle tag
+      if version_req.tag().is_none() && version_req.matches(&resolved.version) {
+        return None;
+      }
+    }
+    Some(version_req)
+  } else {
+    let preferred = if update_to_latest {
+      latest_versions.latest.as_ref()?
+    } else {
+      latest_versions.semver_compatible.as_ref()?
+    };
+    if preferred.version <= resolved?.version {
+      return None;
+    }
+    Some(
+      VersionReq::parse_from_specifier(
+        format!("^{}", preferred.version).as_str(),
+      )
+      .unwrap(),
+    )
+  }
+}
+
+async fn update(
+  mut deps: DepManager,
+  update_to_latest: bool,
+  filter_set: &filter::FilterSet,
+  flags: Arc<Flags>,
+) -> Result<(), AnyError> {
+  let mut updated = Vec::new();
+
+  for (dep_id, resolved, latest_versions) in deps
+    .deps_with_resolved_latest_versions()
+    .into_iter()
+    .collect::<Vec<_>>()
+  {
+    let dep = deps.get_dep(dep_id);
+    let new_version_req = choose_new_version_req(
+      dep,
+      resolved.as_ref(),
+      &latest_versions,
+      update_to_latest,
+      filter_set,
+    );
+    let Some(new_version_req) = new_version_req else {
+      continue;
+    };
+
+    updated.push((dep_id, dep.prefixed_req(), new_version_req.clone()));
+
+    deps.update_dep(dep_id, new_version_req);
+  }
+
+  deps.commit_changes()?;
+
+  if !updated.is_empty() {
+    let factory = super::npm_install_after_modification(
+      flags.clone(),
+      Some(deps.jsr_fetch_resolver.clone()),
+    )
+    .await?;
+
+    let mut updated_to_versions = HashSet::new();
+    let cli_options = factory.cli_options()?;
+    let args = dep_manager_args(
+      &factory,
+      cli_options,
+      deps.npm_fetch_resolver.clone(),
+      deps.jsr_fetch_resolver.clone(),
+    )
+    .await?;
+
+    let mut deps = deps.reloaded_after_modification(args);
+    deps.resolve_current_versions().await?;
+    for (dep_id, prefixed_old_req, new_version_req) in updated {
+      if let Some(nv) = deps.resolved_version(dep_id) {
+        updated_to_versions.insert((prefixed_old_req, nv.version.clone()));
+      } else {
+        log::warn!(
+          "Failed to resolve version for new version requirement: {} -> {}",
+          prefixed_old_req,
+          new_version_req
+        );
+      }
+    }
+
+    log::info!(
+      "Updated {} dependenc{}:",
+      updated_to_versions.len(),
+      if updated_to_versions.len() == 1 {
+        "y"
+      } else {
+        "ies"
+      }
+    );
```

Comment:
I think this output should be improved, it's way less readable than the table in `deno outdated`. 

Also I'm thinking that we might want to print some information before we start downloading packages - I believe we have all the information before we actually start downloading.

Reply:
Yeah it's really barebones currently. Open to suggestions, just a table didn't feel quite right for this